File: api-tester-java/src/main/java/com/revenuecat/apitesterjava/StoreProductAPI.java
package com.revenuecat.apitesterjava;

import com.revenuecat.purchases.ProductType;
import com.revenuecat.purchases.models.StoreProduct;

import org.json.JSONObject;

@SuppressWarnings({"unused"})
final class StoreProductAPI {
    static void check(final StoreProduct product) {
        final String sku = product.getSku();
        final ProductType type = product.getType();
        final String price = product.getPrice();
        final long priceAmountMicros = product.getPriceAmountMicros();
        final String priceCurrencyCode = product.getPriceCurrencyCode();
        final String originalPrice = product.getOriginalPrice();
        final long originalPriceAmountMicros = product.getOriginalPriceAmountMicros();
        final String title = product.getTitle();
        final String description = product.getDescription();
        final String subscriptionPeriod = product.getSubscriptionPeriod();
        final String freeTrialPeriod = product.getFreeTrialPeriod();
        final String introductoryPrice = product.getIntroductoryPrice();
        final long introductoryPriceAmountMicros = product.getIntroductoryPriceAmountMicros();
        final String introductoryPricePeriod = product.getIntroductoryPricePeriod();
        final int introductoryPriceCycles = product.getIntroductoryPriceCycles();
        final String iconUrl = product.getIconUrl();
        final JSONObject originalJson = product.getOriginalJson();
    }

    static void check(final ProductType type) {
        switch (type) {
            case SUBS:
            case INAPP:
            case UNKNOWN:
        }
    }
}

File: api-tester-kotlin/src/main/java/com/revenuecat/apitesterkotlin/StoreProductAPI.kt
package com.revenuecat.apitesterkotlin

import com.revenuecat.purchases.ProductType
import com.revenuecat.purchases.models.StoreProduct
import org.json.JSONObject

@Suppress("unused")
private class StoreProductAPI {
    fun check(product: StoreProduct) {
        val sku: String = product.sku
        val type: ProductType = product.type
        val price: String = product.price
        val priceAmountMicros: Long = product.priceAmountMicros
        val priceCurrencyCode: String = product.priceCurrencyCode
        val originalPrice: String? = product.originalPrice
        val originalPriceAmountMicros: Long = product.originalPriceAmountMicros
        val title: String = product.title
        val description: String = product.description
        val subscriptionPeriod: String? = product.subscriptionPeriod
        val freeTrialPeriod: String? = product.freeTrialPeriod
        val introductoryPrice: String? = product.introductoryPrice
        val introductoryPriceAmountMicros: Long = product.introductoryPriceAmountMicros
        val introductoryPricePeriod: String? = product.introductoryPricePeriod
        val introductoryPriceCycles: Int = product.introductoryPriceCycles
        val iconUrl: String = product.iconUrl
        val originalJson: JSONObject = product.originalJson
    }

    fun check(type: ProductType) {
        when (type) {
            ProductType.SUBS,
            ProductType.INAPP,
            ProductType.UNKNOWN
            -> {}
        }
    }
}
